connect/mirror/src/main/java/org/apache/kafka/connect/mirror/MirrorCheckpointTask.java

```diff
@@ -190,4 +210,59 @@ public void commitRecord(SourceRecord record) {
             Checkpoint.unwrapGroup(record.sourcePartition()),
             System.currentTimeMillis() - record.timestamp());
     }
+
+    private void syncGroupOffset() throws InterruptedException {
+
+        for (String group : consumerGroups) {
+            try {
+                List<Checkpoint> checkpoints = listConsumerGroupOffsets(group).entrySet().stream()
+                    .filter(x -> shouldCheckpointTopic(x.getKey().topic()))
+                    .map(x -> checkpoint(group, x.getKey(), x.getValue()))
+                    .filter(x -> x.downstreamOffset() > 0)  // ignore offsets we cannot translate accurately
+                    .collect(Collectors.toList());
+
+                DescribeConsumerGroupsResult describedGroups = targetAdminClient.describeConsumerGroups(
+                                                                       Lists.newArrayList(group));
+                for (Map.Entry<String, KafkaFuture<ConsumerGroupDescription>> entry :
+                    describedGroups.describedGroups().entrySet()) {
+                    ConsumerGroupDescription consumerGroupDescription = entry.getValue().get();
+                    // consider to sync offset to the target cluster, only if the consumer group is not active
+                    if (consumerGroupDescription.state().equals(ConsumerGroupState.EMPTY)) {
+                        alterGroupOffset(group, checkpoints);
+                    }
+                }
+            } catch (ExecutionException e) {
+                log.error("Error querying offsets for consumer group {} on cluster {}.", group,
+                          sourceClusterAlias, e);
+            }
+        }
+    }
+
+    // Given a consumer group, alter the group consumer offsets in the target cluster,
+    // if and only if the consumer offsets from target cluster are always lower than the translated offsets from upstream
+    private void alterGroupOffset(String group, List<Checkpoint> checkpoints) throws InterruptedException, ExecutionException
+        {
+            Map<TopicPartition, OffsetAndMetadata> offset = new HashMap<>();
+            for (Checkpoint checkpoint : checkpoints) {
+                offset.put(checkpoint.topicPartition(), checkpoint.offsetAndMetadata());
+            }
+
+            boolean sync = true;
+
+            for (Entry<TopicPartition, OffsetAndMetadata> entry :
+                targetAdminClient.listConsumerGroupOffsets(group).partitionsToOffsetAndMetadata()
+                                 .get().entrySet()) {
+                long offsetFromTarget = entry.getValue().offset();
+                long translatedOffset = offset.get(entry.getKey()).offset();
+                // do not alter the entire consumer offsets if any translated offset from upstream
+                // is smaller than the current consumer offset in the target, per partition
```

Comment:
Why skip the entire checkpoint if a single downstream partition is ahead of the checkpoint? I guess that's the safest approach -- but are we sure we can't just skip that partition and write the rest?

Reply:
> but are we sure we can't just skip that partition and write the rest?
the latest version will skip such kind of partition and write the rest